src/libsyntax/test.rs

```diff
@@ -297,11 +297,8 @@ fn is_test_fn(cx: &TestCtxt, i: &ast::Item) -> bool {
         match &i.node {
           &ast::ItemFn(ref decl, _, _, ref generics, _) => {
             let no_output = match decl.output {
-                ast::Return(ref ret_ty) => match ret_ty.node {
```

Comment:
It was intentional, but that is an unfortunate side effect. It can be fixed on quickcheck's end. Whether it needs to be I am less sure, rustc certainly could be more accepting here. Probably both should be fixed.